src/transformStateWithClones.js

```diff
@@ -8,6 +8,35 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const currentState = { ...state };
+  const stateHistory = [];
+
+  for (const obj of actions) {
+    if (obj.type === 'addProperties') {
+      for (const date in obj.extraData) {
+        currentState[date] = obj.extraData[date];
+      }
```

Comment:
Think of how you can use here Object.assign instead of loop

Reply:
thank you! Corrected